### DISCUSSION TITLE
Get toolbar programmatically

### DISCUSSION THREAD

**Discussion by ronaldohoch** [GrapesJS/grapesjs#5043] [Q&A] [answered]
Hello Arthur and Community!

Every time i need to create a new component with a custom toolbar icon, i need to recreate the toolbar this way:

```js
let defaultToolbar = [
        {
          attributes: {
            class: 'fa fa-arrow-up'
          },
          command: e => e.runCommand('core:component-exit', {
            force: 1
          })
        },
        {
          attributes: {
            class: 'fa fa-arrows gjs-no-touch-actions',
            draggable: true
          },
          command: 'tlb-move'
        },
        {
          attributes: {
            class: 'fa fa-clone'
          },
          command: 'tlb-clone'
        },
        {
          attributes: {
            class: 'fa fa-trash-o'
          },
          command: 'tlb-delete'
        }]
```

And then add a new icon anywhere in the code...

Tried this already:
```js
editor.DomComponents.getComponent("image").get("toolbar")
//[]
```
```js
editor.DomComponents.getType("image").model.prototype.defaults.toolbar
//null
```

It would be nice to have a way to get the default toolbar.

**Comment by artf** [comment, marked as answer]
Hi @ronaldohoch the toolbar is created dynamically on component initialisation that's the reason why it's not available in defaults. So if you're defining a new component and you want to edit its toolbar, you can update it on the `init` or extend the default `initToolbar` method. Here you can see an [example](https://github.com/GrapesJS/tui-image-editor/blob/e514a0c2b425d0d36b35cc59e550a7951fdf3ff1/src/index.ts#L176-L193)

**Reply by ronaldohoch** [reply]
Nice, thank you!
